src/hotspot/share/prims/jvmtiEnv.cpp

```diff
@@ -3785,6 +3786,30 @@ JvmtiEnv::GetTime(jlong* nanos_ptr) {
 } /* end GetTime */
 
 
+// info_ptr - pre-checked for null
+jvmtiError
+JvmtiEnv::GetGCCpuTimerInfo(jvmtiTimerInfo* info_ptr) {
+  os::thread_cpu_time_info(info_ptr);
+  return JVMTI_ERROR_NONE;
+} /* end GetGCCpuTimerInfo */
+
+
+// nanos_ptr - pre-checked for null
+jvmtiError
+JvmtiEnv::GetTotalGCCpuTime(jlong* nanos_ptr) {
+  {
+    MutexLocker hl(Heap_lock);
+    if (!os::is_thread_cpu_time_supported() ||
```

Comment:
If it isn't supported then you can't have the capability and so won't reach here.